app/controllers/matches_controller.rb

```diff
@@ -105,7 +107,7 @@ def set_form_dropdowns
 
       @communication_logs = CommunicationLog.where(match: @match)
 
-      @edit_connection_mode = YAML.load(params[:edit_connection_mode].to_s)
+      @edit_connection_mode = params[:edit_connection_mode].to_boolean
```

Comment:
There are a number of places where the existing implementation called `.to_s`. I think this was unnecessary, but i might be missing something?

Reply:
I think if we do a sanity check by stepping through it in multiple browsers, that's enough for me to make sure